src/ziggurat/config.clj

```diff
@@ -108,3 +110,90 @@
 (defn -get [^String key]
   (let [config-vals (get config (keyword key))]
     (java-response config-vals)))
+
+(def ^:private consumer-config-mapping-table
+  {:auto-offset-reset-config        :auto-offset-reset
+   :commit-interval-ms              :auto-commit-interval-ms
+   :consumer-group-id               :group-id
+   :default-api-timeout-ms-config   :default-api-timeout-ms
+   :key-deserializer-class-config   :key-deserializer
+   :session-timeout-ms-config       :session-timeout-ms
+   :value-deserializer-class-config :value-deserializer})
+
+(def ^:private producer-config-mapping-table
+  {:key-serializer-class   :key-serializer
+   :retries-config         :retries
+   :value-serializer-class :value-serializer})
+
+(def ^:private streams-config-mapping-table
+  {:auto-offset-reset-config           :auto-offset-reset
+   :default-api-timeout-ms-config      :default-api-timeout-ms
+   :changelog-topic-replication-factor :replication-factor
+   :session-timeout-ms-config          :session-timeout-ms
+   :stream-threads-count               :num-stream-threads})
+
+(def config-mapping-table
+  (merge consumer-config-mapping-table
+         producer-config-mapping-table
+         streams-config-mapping-table))
+
+(def ^:private non-kafka-config-keys
+  [:channels
+   :consumer-type
+   :input-topics
+   :join-cfg
+   :oldest-processed-message-in-s
+   :origin-topic
+   :poll-timeout-ms-config
+   :producer
+   :thread-count])
+
+(defn- to-list
+  [s]
+  (if (empty? s)
+    (list)
+    (list s)))
+
+(defn- to-string-key
+  [mt k]
+  (-> (get mt k k)
+      (name)
+      (str/replace #"-" ".")))
+
+(defn- normalize-value
+  [v]
+  (str/trim
+   (cond
+     (or (boolean? v) (instance? Long v)) (str v)
+     (keyword? v)                         (name v)
+     :else                                v)))
+
+(defn- set-property
+  [mt p k v]
+  (when-not (some #(= k %) non-kafka-config-keys)
+    (let [string-key (to-list (to-string-key mt k))
+          norm-value (to-list (normalize-value v))]
+      (doseq [sk string-key
+              nv norm-value]
+        (.setProperty p sk nv))))
+  p)
+
+(def ^:private set-all-property (partial set-property config-mapping-table))
+
+(def ^:private set-consumer-property (partial set-property consumer-config-mapping-table))
+
+(def ^:private set-producer-property (partial set-property producer-config-mapping-table))
+
+(def ^:private set-stream-property (partial set-property streams-config-mapping-table))
+
+(defn- build-properties
+  [set-property-fn m]
+  (reduce-kv set-property-fn (Properties.) m))
+
+(def build-all-config-properties (partial build-properties set-all-property))
```

Comment:
This is getting used only in tests so we should move this to the test file. Same for `set-all-property`

Reply:
set property is private

Reply:
No, I mean the above methods are not getting used in the code. We can move them to tests because that's the only place these are getting used.

Reply:
If I have to move this function, I have to also make `set-property` and `mapping-table` to public